h/services/search_index/_queue.py

```diff
@@ -93,10 +99,16 @@ def sync(self, limit):
             return
 
         annotation_ids = {
-            URLSafeUUID.hex_to_url_safe(job.kwargs["annotation_id"]) for job in jobs
+            URLSafeUUID.hex_to_url_safe(job.kwargs["annotation_id"])
+            for job in jobs
+            if not job.kwargs.get("force", False)
         }
-        annotations_from_db = self._get_annotations_from_db(annotation_ids)
-        annotations_from_es = self._get_annotations_from_es(annotation_ids)
+        if annotation_ids:
+            annotations_from_db = self._get_annotations_from_db(annotation_ids)
+            annotations_from_es = self._get_annotations_from_es(annotation_ids)
+        else:
+            annotations_from_db = {}
+            annotations_from_es = {}
```

Comment:
There's no point in reading the annotation from Postgres or Elastic if `force` is `True`, we aren't going to use it. If all of the jobs have `force=True` then there's no point in making a Postgres or Elastic query at all.

Reply:
This actually fits together really nicely. I'm not sure why I thought it would be a good idea to treat them differently.

The only reason I can think is that the performance might be quite different. On the whole I bet we don't issue many writes when doing the normal run of the mill, but we could go absolutely crackers in the case of a big forced index.

This is actually quite easy to remedy if we ever found it to be a real problem. We could have a separate limit for writes. 

So for example, we could do our scan of 1000, find that all 1000 want to be written, know from experience it's only going to work in time with 750 say, and then just bail out. The next iteration would pick the jobs up (because we wouldn't mark them as done).

Long story short, I doubt it matters unless we end up cranking the number of reads up a lot (say we started doing 10,000 at a time because it was very efficient, and we expect to do an average of 1 write) and it's easy to fix if we want to.